### ISSUE TITLE
Recent changes in the "Data_Fn" function results in a message error when using ObsModel_ez

### ISSUE THREAD

**Issue by agruss2** [James-Thorson/VAST#114]
@James-Thorson 
I tried to run some tests for the combined dataset study today.
When trying to build the TmbData object, I obtained the following error message:
```
Error in Data_Fn(Version = Version, FieldConfig = FieldConfig, OverdispersionConfig = OverdispersionConfig,  : 
  objet 'ObsModel' introuvable
```
I suspect that the recent changes in the "Data_Fn" function (made on July, 12th 2018) now results in a message error when using ObsModel_ez. Could you please fix this issue? Many thanks!

**Comment by James-Thorson** [comment]
I'm not following, could you provide more information and preferably a reproducible example?

**Comment by agruss2** [comment]
The code I am using is identical to the one I used yesterday before you introduced some changes in the "Data_Fn" function (i.e., "Combined data types V4.R", which I sent you yesterday). 
Today, when I reached this line:
```
TmbData = Data_Fn("Version"=Version, "FieldConfig"=FieldConfig, "OverdispersionConfig"=OverdispersionConfig, 
	"RhoConfig"=RhoConfig, "ObsModel_ez"=ObsModel_ez, "c_i"=rep(0,nrow(Data_Geostat)), "b_i"=Data_Geostat[,'Catch_KG'], 
	"a_i"=Data_Geostat[,'AreaSwept_km2'], "v_i"=as.numeric(Data_Geostat[,'Vessel'])-1, "e_i"=Data_Geostat[,'Type']-1, 
	"s_i"=Data_Geostat[,'knot_i']-1, "t_i"=Data_Geostat[,'Year'], "a_xl"=Spatial_List$a_xl, 
	"MeshList"=Spatial_List$MeshList, "GridList"=Spatial_List$GridList, "Method"=Spatial_List$Method, 
	"Options"=Options, "Q_ik"=Q_ik, "CheckForErrors"=FALSE, "Aniso"=Aniso )
```
R indicated that: 
```
Error in Data_Fn(Version = Version, FieldConfig = FieldConfig, OverdispersionConfig = OverdispersionConfig,  : 
  objet 'ObsModel' introuvable
```
Hence, I suspect the changes in the "Data_Fn" function you made yesterday resulted in VAST not being able to deal with ObsModel_ez; in other words, I suspect that VAST now again only recognizes ObsModel, which I am not using since I am using ObsModel_ez. 
Is this clearer? 
If need be, I can produce a minimal example for you.
Please let me know. 
Many thanks!

**Comment by agruss2** [comment]
No, if I replace Data_Fn with VAST::Data_Fn, the problem does not go away. 
Therefore, I guess that I have to wait for you to make a new push to GitHub for my script to work again, correct?

**Comment by James-Thorson** [comment]
what happens if you type `args(Data_Fn)` in your R terminal?  I still don't understand what's happening but don't have the same problem on my machine.

**Comment by agruss2** [comment]
If I type args(Data_Fn) in my R terminal, I get:
```
function (Version, FieldConfig, OverdispersionConfig = c(eta1 = 0, 
    eta2 = 0), ObsModel_ez = c(PosDist = 1, Link = 0), VamConfig = c(Method = 0, 
    Rank = 0, Timing = 0), b_i, a_i, c_iz, s_i, t_iz, a_xl, MeshList, 
    GridList, Method, v_i = rep(0, length(b_i)), e_i = c_iz[, 
        1], PredTF_i = rep(0, length(b_i)), X_xj = NULL, X_xtp = NULL, 
    Q_ik = NULL, Aniso = 1, Network_sz = NULL, RhoConfig = c(Beta1 = 0, 
        Beta2 = 0, Epsilon1 = 0, Epsilon2 = 0), t_yz = NULL, 
    CheckForErrors = TRUE, yearbounds_zz = NULL, Options = c(SD_site_logdensity = 0, 
        Calculate_Range = 0, Calculate_effective_area = 0, Calculate_Cov_SE = 0, 
        Calculate_Synchrony = 0, Calculate_proportion = 0)) 
NULL
```

**Comment by James-Thorson** [comment]
well, please remove `VAST` and `SpatialDeltaGLMM`, and then re-install the development version `devtools::install_github("james-thorson/VAST/", ref="development")`.  Next, you'll need to remove any place that `SpatialDeltaGLMM::` occurs in your code, e.g., replace `SpatialDeltaGLMM::Prepare_Extrapolation_Data_Fn(...)` with `Prepare_Extrapolation_Data_Fn(...)` because these helper functions now live in `FishStatsUtils` which is installed and loaded when you load `VAST`.  Then try running again, and please tell me if the error remains.

**Comment by agruss2** [comment]
I followed all of your instructions and then re-ran the script. Unfortunately, the errors remain. What should we do next, then?

**Comment by James-Thorson** [comment]
Please try reproducing that issue on a new machine. I'd you can, then
please send me a minimal example following my standard format. If you
can't, then I'll look at your machine in person.

**Comment by agruss2** [comment]
@James-Thorson To my mind, the problem comes from the use of "ObsModel" in l. 252 here:
https://github.com/James-Thorson/VAST/blob/master/R/Data_Fn.R 
My understanding is that "ObsModel" should not be used in the "Data_Fn" function - apart, of course, for constructing the "Return" list at the end of "Data_Fn.R"; "ObsModel_ez" should be used instead. 
Please let me know if my reasoning is not clear. 
Could you please fix this if I am correct? Many thanks!
